pykafka/protocol.py

```diff
@@ -125,6 +127,45 @@ def raise_error(self, err_code, response):
                 clsname, response))
 
 
+class MessageOffset(object):
+    """A message offset described as a pair:
+    the offset of the message or its containing compressed message,
+    and the offset of the submessage in its containing message, if any."""
+    __slots__ = ("main", "sub")
+
+    def __init__(self, main_offset, sub_offset=None):
+        self.main = main_offset
+        self.sub = sub_offset
+
+    def can_precede(self, other, compressed_topic):
+        """Can those two pairs follow one another?"""
+        if other.sub is None:
+            return (self.main + 1 == other.main) or (
+                compressed_topic and (self.main + 1 < other.main))
+        else:
+            # Compressed message offset may not follow immediately
+            if self.sub is not None:
+                return (self.main == other.main and self.sub + 1 == other.sub) or (
+                    other.sub == 0 and self.main < other.main)
+            else:
+                return other.sub == 0 and self.main < other.main
```

Comment:
Same question here - this doesn't guarantee sequential offsets, only monotonically increasing ones. Does this logic maybe need to include a check of "`compressed_topic`" (`is_compacted_topic`)?